src/test/java/it/com/atlassian/bitbucket/jenkins/internal/deployments/AbstractDeploymentStatusPosterIT.java

```diff
@@ -0,0 +1,117 @@
+package it.com.atlassian.bitbucket.jenkins.internal.deployments;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketNamedLink;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketRepository;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.DeploymentState;
+import com.atlassian.bitbucket.jenkins.internal.status.BitbucketRevisionAction;
+import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder;
+import hudson.model.ItemGroup;
+import hudson.model.Job;
+import hudson.model.Run;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.BitbucketJenkinsRule;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.BitbucketProxyRule;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.GitHelper;
+import it.com.atlassian.bitbucket.jenkins.internal.fixture.JenkinsProjectHandler;
+import jenkins.branch.MultiBranchProject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import static java.lang.String.format;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Following tests use a real Bitbucket Server and Jenkins instance for integration testing, however, the
+ * deployment status is posted against a stub.
+ */
+public abstract class AbstractDeploymentStatusPosterIT {
```

Comment:
No need to change, see this more as a point of discussion. 
Looking at this class it feels more like a utility type class rather than an abstract Test. I do wonder if it would feel more natural to have a `testClient` or similar. As far as I can tell all methods could be made `static` with trivial changes. (usually just pass in the `URL`, or `repoSlug` type of a affair)